### DISCUSSION TITLE
Promoting AntiMicroX

### DISCUSSION THREAD

**Discussion by pktiuk** [AntiMicroX/antimicrox#217] [General]
### Summary
This thread is focused on discussion about making AntiMicroX a bit more popular and accessible for new users.  
Big part of this issue is just regaining users, staying with old legacy [AntiMicro](https://github.com/AntiMicro/antimicro).

I think AntiMicroX is a very good project which doesn't get deserved attention.  
A lot of work has been done to original AntiMirco, and it could currently without any problems replace original AntiMicro which is still quite popular.  
This would help us by providing more feedback, and it could also attract potential contributors.  
This entire issue is a bit inspired by similar one from [xournal++](https://github.com/xournalpp/xournalpp) GitHub (this project has similar history as a fork of abandoned app)

### Some stats
|Label|AntiMicro|AntiMicroX|
|-------|------------|---------------|
|downloads from GitHub|[link](https://somsubhra.github.io/github-release-stats/?username=antimicro&repository=antimicro&page=1&per_page=5) ![Github All Releases](https://img.shields.io/github/downloads/antimicro/antimicro/total.svg) |[link](https://somsubhra.github.io//github-release-stats/?username=antimicrox&repository=antimicrox&page=1&per_page=5) ![Github All Releases](https://img.shields.io/github/downloads/antimicrox/antimicrox/total.svg)|
|Downloads from SourceForge|[link](https://sourceforge.net/projects/antimicro.mirror/files/stats/timeline) ![Downloads AntiMicro](https://img.shields.io/sourceforge/dm/antimicro.svg)|[link](https://sourceforge.net/projects/antimicrox/files/stats/timeline) ![Downloads AntiMicroX](https://img.shields.io/sourceforge/dm/antimicrox.svg)|

Github stars: [link](https://star-history.com/?ref=producthunt#AntiMicro/antimicro&AntiMicroX/antimicrox&Date)
[![Star History Chart](https://api.star-history.com/svg?repos=AntiMicro/antimicro,AntiMicroX/antimicrox&type=Date)](https://star-history.com/?ref=producthunt#AntiMicro/antimicro&AntiMicroX/antimicrox&Date)

Flathub stats can be found at [this](https://klausenbusk.github.io/flathub-stats/#ref=io.github.antimicrox.antimicrox&interval=infinity&downloadType=installs%2Bupdates) site

### Things to do
In terms of accessibility, there are still some things to do:  
1. Restore Windows support - most of the users of AntiMicro are Windows users. https://github.com/AntiMicroX/antimicrox/issues/4
2. Make it easier to find for potential users - it would be good if AntiMicroX would be easier to find by someone who is looking for this type of software. We have currently added AntiMicroX to: [AlternativeTo](https://alternativeto.net/software/antimicrox/) and [Sourceforge](https://sourceforge.net/projects/antimicrox/).
3. Make it easily accessible for users - in case of Linux builds it would be great to make this app accessible via official repositories and other channels of distribution. 
4. It would be good to have some simple site of project, it could be easily done via Github pages [Our site](https://github.com/AntiMicroX/antimicrox.github.io) - work in progress

What do you think about these tasks? Do you have any comments in terms of accessibility and visibility for new and old users?  
This is also a very good place to share your first impression.

**To sum up:**
If you want to help you can:
- Like AntiMicroX's on [AlternativeTo](https://alternativeto.net/software/antimicrox/) and [SourceForge](https://sourceforge.net/projects/antimicrox/)
- Consider helping us with one of tasks mentioned here (currently Windows port is our top priority)
- Contributions in any form are welcome

**Comment by AriaMoradi** [comment]
- how about adding it to debian? ubuntu? SteamOS? set up a PPA?
- apt has an alternative symbolic link feature(a good example is how they handle bsd's vs gnu's netcat)
- antimicro has wide packaging availability,  contact those repositories' maintainers about antimicro getting deprecated and offer to replace it with antimicrox(antimicrox as an alias of antimcro, which comes back to my second idea)
- setup some articles about the good and new fork here and there.

btw why rename the package? google offers to correct the new name with the old one!(did you mean antimicro?) this kills most of the traffic by itself!

Consider a scenario when someone found antimicro years ago, finds it again by a little search, but they have no good way to find antimicrox.
 I myself where going for https://github.com/Tylemagne/Gopher360 at first but no linux support! alternativeto got me here.

**Comment by pktiuk** [comment]
> how about adding it to debian? ubuntu? SteamOS? set up a PPA?

(I assume you are talking about adding it to official debian repositories. ) 
It isn't submitted to official Debian repositories, because this package is not ready (yet) to be added there, because it doesn't match strict quality requirements required from packages to get there (you can check using `lintian` to check our debs), but we are working on it.  
Moreover, as far as I know (I can be wrong) the process of adding a new package for Debian is less noob-firendly than similar systems for fedora and arch (their [site](https://www.debian.org/devel/wnpp/index.en.html) is also awful and outdated). Everything is done using Debian bug tracking system which is not very friendly for new users.  
Adding package is ubuntu would be also tricky, because they prefer to take packages directly from Debian.  
I was thinking about setting up a PPA, but I haven't checked how much time it would take me to set it up.

> apt has an alternative symbolic link feature

Could you give me some links describing use of it, my attempts of googling it ended at StackOverflow looking at different problems with linking...

> antimicro has wide packaging availability, contact those repositories' maintainers about antimicro getting deprecated and offer to replace it with antimicrox(antimicrox as an alias of antimcro, which comes back to my second idea)

Which repositories do you mean? Do you mean Linux repositories or sites like Sourceforge?

> setup some articles about the good and new fork here and there

It is not very dependent on us, but some sites have noticed our efforts :smile:
https://itsfoss.com/antimicrox/
https://www.reddit.com/r/linux_gaming/comments/ineunz/antimicro_fork_antimicrox_is_a_graphical_program/
http://ubuntuhandbook.org/index.php/2020/10/antimicrox-map-mouse-keyboard-gamepad/

> btw why rename the package?

It was done by the original developer of AntiMicroX, it was probably (I can't say I am sure about it) caused by problems with lack of will to pass this project (already abandoned) to someone else by creators of legacy AntiMicro creators (just look at [these repositories](https://github.com/AntiMicro/antimicro), there is no information about ending development of project, it is still active and people are still reporting bugs there).

**Comment by pktiuk** [comment]
@AriaMoradi  
Thank you for your questions, thank to them I realized there is already antimicro package in debian repositories and I could try to contact its maintainer, maybe they could easily deal with problems we have and could deal with changing status of antimicro as 
deprecated. :rocket:

**Comment by AriaMoradi** [comment]
> Could you give me some links describing use of it, my attempts of googling it ended at StackOverflow looking at different problems with linking...

this feature comes from debian itself.
 docs: 
- https://wiki.debian.org/DebianAlternatives
- http://manpages.ubuntu.com/manpages/trusty/man8/update-alternatives.8.html

> Which repositories do you mean? Do you mean Linux repositories or sites like Sourceforge?

Both. Just google antimicro and check out every link for a few pages. you will be surprised at how many there are out there! mostly windows though...

> (just look at these repositories, there is no information about ending development of project, it is still active and people are still reporting bugs there).

The only way I see is continuing to direct issues over here, maybe breathe down the original devs necks @jsbackus and @juliagoda and @AmenStop to ask them to officially post about this project and their abandoing of it.(4-3-2 years of issues is too much in the world of open source)
**Still** I think there is no reason to change the project's name, it's under GPLv3 anyways(and I can't find any trademarks about the name). This will help with traffic/SEO alot. trust me. maybe what @juliagoda did back then was a good thing, distinguishing the frok from the original in case the original dev came back. Now it's different. four years has passed and that's surely enough.  
edit: **One problem is @juliagoda didn't mention any links this project and unpinned your issue. this is not good.**

> @AriaMoradi
> Thank you for your questions, thank to them I realized there is already antimicro package in debian repositories and I could try to contact its maintainer.

happy to help

**Comment by jsbackus** [comment]
Handing over the project is something I've been meaning to do for a while,
just been to overwhelmed to do even that. If this project wants to take
over, pls contact me via email.

**Comment by AriaMoradi** [comment]
> Handing over the project is something I've been meaning to do for a while,
just been to overwhelmed to do even that. If this project wants to take
over, pls contact me via email.

There you go! now you can take over the original account(and distributions)! Thanks to @jsbackus

**Comment by jsbackus** [comment]
:) just glad to see someone else carrying the torch, and sad that I haven't
been able to find a good home for AntiMicro sooner

**Comment by juliagoda** [comment]
@AriaMoradi 

First of all, please finally stop calling me to the board, because it's already getting distasteful and I get unwanted messages in my email address. Secondly, I was open to everything until a certain point, until pktiuk definitely did not exaggerate in his actions.  I am sorry that I am not perfect like him because I had some bugs or whatever. If my fork is so bad, you didn't have to fork it, you just could take care of the official version and continue it. I don't have to explain myself to anyone from something I do for free, but I continued the program for about 2 years and all I read is statements like yours. 

"maybe what @juliagoda did back then was a good thing" - maybe it's because, for example, you don't have to use this fork? I wanted to continue and learn something from it and I did it. As I said, I have no intention of explaining it to anyone. I'm not going to pin issue of people who play with constant criticism and pretend to know everything better than others. I had pinned the  information, but they abused my courtesy and the discussion is ended. I don't like to hear someone speak when doesn't even know what's going on.

Besides, how can I link to the person who told me that he is creating a continuation of the project only to make it accessible to as many people as possible, but he will not add any new functionality and will not do anything else but marketing. It is very convenient to publicize not your work, sign it, criticize the mistakes of your predecessors while doing little. I will not continue the discussion because it does not make any sense. I do not have time for this.

**Comment by pktiuk** [comment]
People can often disagree about direction of app development in general.  
That's why there are so many forms of multiple applications and systems.  
I understand you have put a lot of hours and effort into this project, but as you stated earlier it was also a way learning, and we all make mistakes when we learn, especially when we do it entirely on our own without anyone to review what we do.

But this time there is also another reason for creating fork. It is also a way of preserving someone else's code from obsolescence. And this was clearly the reason of forking AntiMicroX from AntiMicro, and it is also the reason of forking this repository from your version.

That's why I consider myself more of a maintainer than a developer, which wants to keep this project alive and make it usable for as wide audience as possible.  
I want to apologize for lack of rapid development and groundbreaking changes but it is just my side project which has to share time with my job, university and participation in students associations.

I understand you don't have time, as you told earlier that it was the main reason for stopping development of AntiMicroX. Because of that I haven't mentioned you earlier in issues when I had problems with understanding some changes in code. If you find these notifications annoying disable them here: https://github.com/watching

**Comment by gombosg** [comment]
@pktiuk 
I think these comments are totally on topic about "promoting this software"), let's not delete important discussion history on an open forum, I don't agree with that.

@juliagoda (How can I mention you in a discussion without mentioning you? :slightly_smiling_face:)
I understand your feelings about the project. You have clearly put in a lot of work, which we and all the users appreciate. 

If you didn't archive the repo, we were likely just helping out with that version. Also, if there was another maintained fork of https://github.com/AntiMicro/antimicro or https://github.com/juliagoda/antimicroX, then we would be maintaining that or team up with the maintainers so that only 1 fork exists.

> It is very convenient to publicize not your work, sign it, criticize the mistakes of your predecessors while doing little.

Where have we criticized you? We are **volunteers** and doing it not because/despite any feelings, but because we think it's a fine GPL FOSS project and it should be distributed to users. Many FOSS projects are in maintenance mode, but we fixed bugs and builds, created packages and kept the codebase in good standing which I think is really useful for users. 

I understand your criticism that this is "just maintenance" but even that takes several hours of volunteer work.

> I had pinned the information, but they abused my courtesy...

We apologize. Is it possible for you to forgive what we have done, and do what's best for the project and the users: put a link to our repo in your README so that they will be able to find the only maintained fork?

**Comment by AriaMoradi** [comment]
> First of all, please finally stop calling me to the board, because it's already getting distasteful and I get unwanted messages in my email address. Secondly, I was open to everything until a certain point, until pktiuk definitely did not exaggerate in his actions. I am sorry that I am not perfect like him because I had some bugs or whatever. If my fork is so bad, you didn't have to fork it, you just could take care of the official version and continue it. I don't have to explain myself to anyone from something I do for free, but I continued the program for about 2 years and all I read is statements like yours.
> 
> "maybe what @juliagoda  did back then was a good thing" - maybe it's because, for example, you don't have to use this fork? I wanted to continue and learn something from it and I did it. As I said, I have no intention of explaining it to anyone. I'm not going to pin issue of people who play with constant criticism and pretend to know everything better than others. I had pinned the information, but they abused my courtesy and the discussion is ended. I don't like to hear someone speak when doesn't even know what's going on.
> 
> Besides, how can I link to the person who told me that he is creating a continuation of the project only to make it accessible to as many people as possible, but he will not add any new functionality and will not do anything else but marketing. It is very convenient to publicize not your work, sign it, criticize the mistakes of your predecessors while doing little. I will not continue the discussion because it does not make any sense. I do not have time for this.

First I should apologize if my wording caused any misunderstanding. I didn't intend to discredit or criticize your work, I was just analyzing the current status quo and calling it "not good" meant that since it's read-only now, it leaves the potential users left out in the dark. 
As I said before the only way for a great project like this to live and grow is if it gains traction, and yes we as a community using this software should market the hell out of it(as any other ArchLinux user like me does market Arch that it has become a meme!). 
Second for a project as stable and feature packed as this one, maintaining comes first. And I'm sure your work was good that they forked it, if it wasn't they wouldn't bother. And sorry but I am not aware of the ins and outs of the code and which features are included or removed from your repo; I'm just a new user offering their insights and trying to help. what happens next will be decided by the current team as it is everywhere else.

**Comment by pktiuk** [comment]
@AriaMoradi 
In terms of renaming after statement of jsbackus we will have to rethink our project.  
Maybe we could really go upstream and use changes from AntiMicroX in legacy AntiMicro.

We will have to think how to deal with this possibility, because It won't as simple as pushing changes on new repo, it will also require updating our packages for fedora, arch and flathub.

**Comment by AriaMoradi** [comment]
> @AriaMoradi
> In terms of renaming after statement of jsbackus we will have to rethink our project.
> Maybe we could really go upstream and use changes from AntiMicroX in legacy AntiMicro.
> 
> We will have to think how to deal with this possibility, because It won't as simple as pushing changes on new repo, it will also require updating our packages for fedora, arch and flathub.

(since renaming the repo is related to promoting I'll write my thoughts here, if not open a new issue for it)

### best case scenario
1- pull from the old project, put the current branches under legacy named ones(i.e. master -> master_legacy)
2- force push and overwrite Antimicro/antimicro with what we have got
3- maybe bump a major version?
4- distribute both versions(antimicro and antimicro-legacy)
5- remove the legacy version when we are sure antimicrox is a strict superset of the old antimicro in respect to features


### less desirable scenario
1- restore windows functionality and other features if removed from legacy 
2- make sure antimicrox is a strict superset of the old antimicro in respect to features
3- force push and overwrite Antimicro/antimicro with what we have got
4- package and distribute everywhere
5- maybe also distribute the old code as antimicro-legacy?

**Comment by pktiuk** [comment]
Firstly, I will try to go with AntiMicroX releases as possible without removing Windows (its support wasn't removed after first release).  
But it would require setting up windows on my machine (I really dislike this system).  
It would be great to have CI building on Windows.

**Comment by jsbackus** [comment]
With AntiMicro, I was able to test it under Wine. You might have trouble testing for bugs, at least in programs that don't work in Wine, but you might try using it for testing. Also, leverage the community. If no one is willing to step up to help with testing, then I wouldn't worry about adding support back in :smile: 

Also - I highly recommend switching the Windows installer from Wix to something like NSIS that works pretty well under Wine.

**Comment by pktiuk** [comment]
@AriaMoradi   
Would you like to join us and start working for the glory of F/OSS?

**Comment by AriaMoradi** [comment]
> @AriaMoradi
> Would you like to join us and start working for the glory of F/OSS?

I'm not good at c/c+ and other libs used here but I will help where I can(from the shadows if you will)

**Comment by pktiuk** [comment]
@AriaMoradi  
Currently, I am the only one using and knowing C++ in this team ;)

**Comment by ArtechFuz3D** [comment]
Hello I would like to contribute to this project. I find it incredibly useful, currently mapping out a suite of control sets for blender controls. I use Linux mostly.

**Reply by pktiuk** [reply]
I'm glad to hear that.

If you are familiar with C++ you could help us with fixing some of our bugs or implementing some functionalities.  
If you are a bit more into web development you could help us finish our [half-baked website](https://github.com/AntiMicroX/antimicrox.github.io).    
You can also [help with translation](https://github.com/AntiMicroX/antimicrox/wiki/Translating-AntiMicroX)

What would you like to do?

**Reply by ArtechFuz3D** [reply]
I don't know C unfortunately. Id be happy to help on the website, shall we start there?

**Reply by avinal** [reply]
Hey, @clickonrefresh, great choice. here is our website: https://github.com/AntiMicroX/antimicrox.github.io, it is created using vuepress. Take a look please and let us know. There are a few issues too that will help you know what is to be expected in the website.

**Reply by pktiuk** [reply]
I sent you an invitation to our team on GitHub

**Reply by ArtechFuz3D** [reply]
Thanks glad to be here. I had a look at the issues for website, I havnt used vuepress yet so will just take a moment to familiarize myself. Looks straight forward though.
